airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/streams.py

```diff
@@ -171,22 +180,30 @@ def request_params(self, stream_state: Mapping[str, Any], **kwargs) -> MutableMa
 
     def _state_filter(self, stream_state: Mapping[str, Any]) -> Mapping[str, Any]:
         """Additional filters associated with state if any set"""
-        state_value = stream_state.get(self.cursor_field)
-        filter_value = self._start_date if not state_value else pendulum.parse(state_value)
 
         potentially_new_records_in_the_past = self._include_deleted and not stream_state.get("include_deleted", False)
         if potentially_new_records_in_the_past:
             self.logger.info(f"Ignoring bookmark for {self.name} because of enabled `include_deleted` option")
-            filter_value = self._start_date
+
+        def _single_state_filter(single_stream_state: Mapping[str, Any]):
+            state_value = single_stream_state.get(self.cursor_field)
+            filter_value = self._start_date if not state_value else pendulum.parse(state_value)
+
+            if potentially_new_records_in_the_past:
+                filter_value = self._start_date
+
+            return {
+                "field": f"{self.entity_prefix}.{self.cursor_field}",
+                "operator": "GREATER_THAN",
+                "value": filter_value.int_timestamp,
+            }
+
+        result_filtering_custom = {}
+        for account_stream_state in stream_state.keys():
+            result_filtering_custom[account_stream_state] = _single_state_filter(stream_state[account_stream_state])
```

Comment:
please don't use nested functions, it is a very bad practice, this affects readability and testing

Reply:
also could please ensure that any function has a docstring explaining why do we need it